DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,360 @@
+﻿// ReSharper disable InconsistentNaming
+
+namespace DigitalLearningSolutions.Data.Migrations
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using System.Transactions;
+    using Dapper;
+    using FluentMigrator;
+    using Microsoft.Data.SqlClient;
+
+    [Migration(202204071530, TransactionBehavior.None)]
+    public class PopulateUsersTableFromAccountsTables : Migration
+    {
+        public override void Up()
+        {
+            IDbConnection connection = new SqlConnection(ConnectionString);
+
+            var options = new TransactionOptions
+            {
+                Timeout = new TimeSpan(0, 15, 0),
+            };
+            using var transactionScope = new TransactionScope(TransactionScopeOption.Required, options);
+
+            // 1. Delete from Users (this should be empty)
+            connection.Execute("Delete Users");
+
+            // 2. Copy AdminAccounts to Users table
+            connection.Execute(
+                @"INSERT INTO dbo.Users (
+                    PrimaryEmail,
+                    PasswordHash,
+                    FirstName,
+                    LastName,
+                    JobGroupID,
+                    ProfessionalRegistrationNumber,
+                    ProfileImage,
+                    Active,
+                    ResetPasswordID,
+                    TermsAgreed,
+                    FailedLoginCount,
+                    HasBeenPromptedForPrn,
+                    LearningHubAuthId,
+                    HasDismissedLhLoginWarning,
+                    EmailVerified
+                )
+                SELECT Email,
+                    Password_deprecated,
+                    Forename_deprecated,
+                    Surname_deprecated,
+                    10,
+                    NULL,
+                    ProfileImage_deprecated,
+                    Active,
+                    ResetPasswordID_deprecated,
+                    TCAgreed_deprecated,
+                    FailedLoginCount_deprecated,
+                    0,
+                    NULL,
+                    0,
+                    GETUTCDATE()
+                    FROM AdminAccounts"
+            );
+
+            // 3. Update AdminAccounts to reference Users.ID
+            connection.Execute(
+                @"UPDATE AdminAccounts
+                SET UserID = (SELECT ID FROM Users WHERE Email = PrimaryEmail)"
+            );
+
+            // 4. DelegateAccount

```

Comment:
This probably doesn't match the algorithm I laid out in the ticket (sorry!), but to shortcut pulling the vast majority of our delegates into memory and making individual queries for them, can we start the DelegateAccount step with a query which copies delegates with new (and unique) emails directly into Users? Then we can forget about these for the rest of the migration.